app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -0,0 +1,40 @@
+module OpenidConnect
+  class AuthorizationController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def index
+      @authorize_form = OpenidConnectAuthorizeForm.new(params)
+
+      success = @authorize_form.valid?
+      analytics.track_event(Analytics::OPENID_CONNECT_REQUEST_AUTHORIZATION,
+                            success: success,
+                            client_id: @authorize_form.client_id,
+                            errors: @authorize_form.errors.full_messages)
+      if success
+        render :index
+      else
+        render :error
+      end
+    end
+
+    def create
+      @authorize_form = OpenidConnectAuthorizeForm.new(params)
+      redirect_uri = @authorize_form.submit(current_user)
+
+      if redirect_uri
+        analytics.track_event(Analytics::OPENID_CONNECT_ALLOW, client_id: @authorize_form.client_id)
+        redirect_to redirect_uri
+        return
+      end
+
+      render :error
+    end
+
+    def destroy
+      @authorize_form = OpenidConnectAuthorizeForm.new(params)
+      analytics.track_event(Analytics::OPENID_CONNECT_DENY, client_id: @authorize_form.client_id)
```

Comment:
This is a user-initiated event, right? If the user chooses to decline, should we rename this event to `OPENID_CONNECT_USER_DECLINED`, or similar? `Deny` makes it sound like we denied the user because of something that went wrong, as opposed to a deliberate choice they made.